src/main/scala/org/broadinstitute/dsde/workbench/sam/service/ResourceService.scala

```diff
@@ -203,6 +203,21 @@ class ResourceService(private val resourceTypes: Map[ResourceTypeName, ResourceT
     }
   }
 
+  /**
+    * Overwrites an existing policy's membership (keyed by resourceType/resourceId/policyName) if it exists
+    * @param resourceType
+    * @param policyName
+    * @param resource
+    * @param membersList
+    * @return
+    */
+  def overwritePolicyMembers(resourceType: ResourceType, policyName: AccessPolicyName, resource: Resource, membersList: Set[WorkbenchEmail]): Future[AccessPolicy] = {
+    loadResourcePolicy(ResourceAndPolicyName(resource, policyName)).flatMap {
+      case None => throw new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.NotFound, "Policy not found"))
```

Comment:
just fyi, you can `Future.failed(new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.NotFound, "Policy not found")))` instead of throwing the exception

Reply:
not necessarily disagreeing, but why should throwing an exception be avoided?

Reply:
1. throwing exception with stack trace is expensive
2. throwing exception makes code less compose-able in general.

For this particular code, unless caller for this function wraps it inside a `Try` or `Future`, this will actually crash the app? (if we use `Future.failed(some exception)`, then we don't need to worry about this at all)

Reply:
I would not let the expense sway us. Of course throwing exceptions in the normal code path is bad. Used in exceptional situations the expense does not amount to much.

Thinking about thrown exceptions as a side effect though is a good argument. Side effects are bad. Since we use some java libs too, we will never be without. Also, when in Future or Try blocks it seems to be the only way to signal failure.

In this particular case it doesn't really matter since the throw is within a flatMap of a Future. But since it is a flatMap, it is more correct to use Future.failed